### PR TITLE
LibPDF: Fix sign (and fixed point) in glyph decoding opcode 24

### PR DESCRIPTION
**Pull request by nico**
Two bugs:

1. We decoded a u32, not an i32 as the spec wants
2. (minor) Our fixed-point divisor was off by one

Fixes text rendering in Bakke2010a.pdf in pdffiles, and rendering of other fonts with negative width adjustments from optcode 255. That PDF was produced by "Apple pstopdf" and uses font SFBX1200, which is apparently a variant of Computer Modern. So maybe this helps with lots of PDFs produced from TeX files, but I haven't checked that.

### CODE REVIEW

**Comment by nico** [comment]
Before:

Screenshot 2023-10-15 at 9 49 13 AM

After:

Screenshot 2023-10-15 at 9 49 02 AM

**Comment by nico** [comment]
Red CI is unrelated.

File: Userland/Libraries/LibPDF/Fonts/Type1FontProgram.cpp [hunk 1]
```diff
@@ -250,16 +253,17 @@ PDFErrorOr<Type1FontProgram::Glyph> Type1FontProgram::parse_glyph(ReadonlyBytes
...
         if (v == 255) {
             TRY(require(4));
-            int a = data[++i];
-            int b = data[++i];
-            int c = data[++i];
-            int d = data[++i];
+            // Both Type 1 and Type 2 spec:
+            // "If the charstring byte contains the value 255, the next four bytes indicate a two’s complement signed number.
+            //  The first of these four bytes contains the highest order bits [...]"
+            i32 a = static_cast<i32>((data[i + 1] << 24) | (data[i + 2] << 16) | (data[i + 3] << 8) | data[i + 4]);
```
Comment (gmta) [inline Userland/Libraries/LibPDF/Fonts/Type1FontProgram.cpp:259]: Could maybe use `i32 a = BigEndian<i32>(*reinterpret_cast<i32*>(data));` here?

Comment (Hendiadyoin1) [reply Userland/Libraries/LibPDF/Fonts/Type1FontProgram.cpp:259]: Funny way to write `bit_cast`

Comment (nico) [reply Userland/Libraries/LibPDF/Fonts/Type1FontProgram.cpp:259]: I had that at first, but I liked it less /shrug

Comment (gmta) [reply Userland/Libraries/LibPDF/Fonts/Type1FontProgram.cpp:259]: Personally I like the readability of "take the big endian i32 at this location" vs. having to check byte indices and shift sizes. But feel free to ignore, someone else might do a big sweep over the codebase one day to replace these occurrences anyway :^)

Comment (AtkinsSJ) [reply Userland/Libraries/LibPDF/Fonts/Type1FontProgram.cpp:259]: :yakshed: Maybe we want a `template<typename T> T& Bytes::value_at(offset)` method to make this nicer to do?

Comment (nico) [reply Userland/Libraries/LibPDF/Fonts/Type1FontProgram.cpp:259]: This PR is about fixing a bug in existing code and uses code similar in style to the code the bug is in. Maybe this discussion could happen somewhere that isn't this PR? 😅

File: Userland/Libraries/LibPDF/Fonts/Type1FontProgram.cpp [hunk 2]
```diff
@@ -250,16 +253,17 @@ PDFErrorOr<Type1FontProgram::Glyph> Type1FontProgram::parse_glyph(ReadonlyBytes
...
+            i += 4;
             if (is_type2) {
-                auto integer = float((a << 8) | b);
-                auto fraction = float((c << 8) | d) / AK::NumericLimits<u16>::max();
-                TRY(push(integer + fraction));
-            } else
-                TRY(push((a << 24) + (b << 16) + (c << 8) + d));
```
Comment (nico) [inline Userland/Libraries/LibPDF/Fonts/Type1FontProgram.cpp:262]: (ref)

File: Userland/Libraries/LibPDF/Fonts/Type1FontProgram.cpp (hunk 1 above)
Comment (awesomekling) [reply Userland/Libraries/LibPDF/Fonts/Type1FontProgram.cpp:259]: > Maybe this discussion could happen somewhere that isn't this PR?

Fair.

**Pull request merged**